src/journal.c

```diff
@@ -116,119 +115,119 @@ static const UnkStruct_020E5B50 Unk_020E5B50[] = {
 {
-    memset(journal, 0, sizeof(Journal) * 10);
+    memset(journalEntry, 0, sizeof(JournalEntry) * MAX_JOURNAL_ENTRIES);
 }
 
-Journal *SaveData_GetJournal(SaveData *param0)
+JournalEntry *SaveData_GetJournal(SaveData *saveData)
 {
-    return SaveData_SaveTable(param0, 18);
+    return SaveData_SaveTable(saveData, 18);
 }
 
-Journal *Journal_GetSavedPage(Journal *journal, BOOL param1)
+JournalEntry *Journal_GetSavedPage(JournalEntry *journalEntry, BOOL journalAcquired)
 {
-    RTCDate v0;
-    s32 v1;
+    RTCDate currDate;
+    s32 i;
 
-    if (param1 != 1) {
+    if (journalAcquired != TRUE) {
         return NULL;
     }
 
-    GetCurrentDate(&v0);
+    GetCurrentDate(&currDate);
 
-    if ((journal[0].unk_00.unk_00_7 != 0) && ((journal[0].unk_00.unk_00_0 != v0.year) || (journal[0].unk_00.unk_00_7 != v0.month) || (journal[0].unk_00.unk_00_14 != v0.day) || (journal[0].unk_00.unk_00_11 != v0.week))) {
-        for (v1 = 10 - 1; v1 >= 1; v1--) {
-            journal[v1] = journal[v1 - 1];
+    if ((journalEntry[0].title.month != 0) && ((journalEntry[0].title.year != currDate.year) || (journalEntry[0].title.month != currDate.month) || (journalEntry[0].title.day != currDate.day) || (journalEntry[0].title.week != currDate.week))) {
+        for (i = MAX_JOURNAL_ENTRIES - 1; i >= 1; i--) {
+            journalEntry[i] = journalEntry[i - 1];
         }
 
-        Journal_Init(&journal[0]);
+        Journal_Init(&journalEntry[0]);
     }
 
-    return &journal[0];
+    return &journalEntry[0];
 }
 
-BOOL Journal_CheckOpenOnContinue(Journal *journal, BOOL param1)
+BOOL Journal_CheckOpenOnContinue(JournalEntry *journalEntry, BOOL journalAcquired)
 {
-    RTCDate v0;
-    RTCDate v1;
-    int v2;
+    RTCDate journalEntryTitle;
+    RTCDate currDate;
+    int daysDiff;
 
-    if (param1 != 1) {
-        return 0;
+    if (journalAcquired != TRUE) {
+        return FALSE;
     }
 
-    GetCurrentDate(&v1);
+    GetCurrentDate(&currDate);
 
-    v0.year = journal[0].unk_00.unk_00_0;
-    v0.month = journal[0].unk_00.unk_00_7;
-    v0.day = journal[0].unk_00.unk_00_14;
-    v0.week = journal[0].unk_00.unk_00_11;
+    journalEntryTitle.year = journalEntry[0].title.year;
+    journalEntryTitle.month = journalEntry[0].title.month;
+    journalEntryTitle.day = journalEntry[0].title.day;
+    journalEntryTitle.week = journalEntry[0].title.week;
 
-    v2 = DayNumberForDate(&v1) - DayNumberForDate(&v0);
+    daysDiff = DayNumberForDate(&currDate) - DayNumberForDate(&journalEntryTitle);
 
-    if (((v1.month == 12) && (v1.day == 31) && (v0.month == 1) && (v0.day == 1)) || ((v1.month == 1) && (v1.day == 1) && (v0.month == 12) && (v0.day == 31))) {
-        s32 v3 = (s32)v1.year - (s32)v0.year;
+    if (((currDate.month == 12) && (currDate.day == 31) && (journalEntryTitle.month == 1) && (journalEntryTitle.day == 1)) || ((currDate.month == 1) && (currDate.day == 1) && (journalEntryTitle.month == 12) && (journalEntryTitle.day == 31))) {
```

Comment:
**polish**: Clean up unnecessary parens and split conditionals up onto separate lines for legibility.